runtime/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -753,9 +764,9 @@ private int decodeInitialType(
         if (packet != null)
         {
             final int length = packet.remainingLength();
-            final MqttPacketType packetType = MqttPacketType.valueOf(packet.typeAndFlags() >> 4);
+            final MqttPacketType packetType = MqttPacketType.valueOf((packet.typeAndFlags() >> 4) * 2 + server.typeOffset);
```

Comment:
This approach is too cryptic with the `* 2 + server.typeOffset`, let's revert as it's not needed if we keep the decoders separate by version, as they should be anyway.